fiat-api/src/main/java/com/netflix/spinnaker/fiat/api/Resource.java

```diff
@@ -14,17 +14,25 @@
  * limitations under the License.
  */
 
-package com.netflix.spinnaker.fiat.model.resources;
+package com.netflix.spinnaker.fiat.api;
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.netflix.spinnaker.kork.annotations.Alpha;
+import org.pf4j.ExtensionPoint;
 
-public interface Resource {
+/** Represents a Fiat resource (e.g., a Role or a Service account). */
+@Alpha
+public interface Resource extends ExtensionPoint {
```

Comment:
I had problems deserializing these if they extended `SpinnakerExtensionPoint` and didn't want to make this PR any larger by updating the object mapper. I'll come back to this.

Reply:
The consequences of not using `SpinnakerExtensionPoint` is that the proxy will not work (metrics, logging, etc). I'd be interested to hear more about the serialization issues you're seeing. I suspect it's the fact that `getExtensionClass()` is automatically picked up as a property for Jackson?

I think the right resolution (if my assumption is correct) would be to change `SpinnakerExtensionPoint` to rename `getExtensionClass()` to something that won't be automatically picked up by Jackson.